src/metrics/chaoss.ts

```diff
@@ -358,3 +358,62 @@ FORMAT JSONCompact`;
     }
   });
 };
+
+interface IssueAgeOptions {
+  type: 'avg' | 'median';
+  unit: 'week' | 'day' | 'hour' | 'minute';
+}
+export const chaossIssueAge = async (config: QueryConfig<IssueAgeOptions>) => {
+  config = getMergedConfig(config);
+  const whereClauses: string[] = ["type = 'IssuesEvent'"];
+  const repoWhereClause = getRepoWhereClauseForClickhouse(config);
+  if (repoWhereClause) whereClauses.push(repoWhereClause);
+
+  const endDate = new Date(`${config.endYear}-${config.endMonth}-1`);
+  endDate.setMonth(config.endMonth);
+  
+  let type = filterEnumType(config.options?.type,['avg','median'],'avg');
+  let unit = filterEnumType(config.options?.unit,['week','day','hour','minute'],'day');
+
+  const sql = `
+SELECT 
+  id, 
+  argMax(name, time) AS name,
+  ${getGroupArrayInsertAtClauseForClickhouse(config, {key:`issue_age`, defaultValue: 'NaN'})}
+FROM
+(
+  SELECT
+    ${getGroupTimeAndIdClauseForClickhouse(config, 'repo','opened_at')},
+    round(${type}(dateDiff('${unit}', opened_at, toDate('${endDate.getFullYear()}-${endDate.getMonth()+1}'))),${config.percision}) AS issue_age
```

Comment:
Now the error message is :
`Error: Syntax error: failed at position 7046 ('Jan') (line 9, col 47): Jan 01 2016 00:00:00 GMT+0000 (GMT))),2) AS issue_age  FROM  (    SELECT      repo_id,      argMax(repo_name, created_at) AS repo_name,      org_id,     .....`

I think it's due to the `startdate` and `enddate` parameters for the `dateDiff` function are in different datetime forms. For example, opened_at is something like `2015-01-29 20:11:07`, while the end date is something like `2015-01`

Reply:
Do we have a better solution for this?

Reply:
The problem here is not format or data type difference. Actually the `startdate` and `enddate` can be `Date` or `DateTime`.

The `opened_at` is selected from the `created_at` column so it is `DateTime` type.

You use `toDate` to construct the `enddate` so it will be `Date` type. But it is OK to get `dateDiff` result for a `Date` param and a `DateTime` param.

So the real problem is you should change `toDate('${endDate.getFullYear()}-${endDate.getMonth()+1}')` to `toDate('${endDate.getFullYear()}-${endDate.getMonth()+1}-1')` which will change `2015-1` to `2015-1-1`, so the `toDate` function can understand what you want.